srcpkgs/retroarch/template

```diff
@@ -1,57 +1,83 @@
+ $(vopt_if x11 'libXext-devel libXinerama-devel libXv-devel libXxf86vm-devel')"
 depends="$(vopt_if vulkan 'vulkan-loader')"
 short_desc="Official reference frontend for the libretro API"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="GPL-3.0-or-later"
-homepage="http://www.libretro.com/"
+homepage="https://www.retroarch.com/"
 distfiles="https://github.com/libretro/RetroArch/archive/v$version.tar.gz"
-checksum=027e9eb243488da1b8532116eab2a79932fcf2dcb96bcc4be9a12b329a2a9c52
-build_options="ffmpeg opengl jack pulseaudio sdl2 x11 vulkan qt5"
+checksum=39d7ad9298c9487fb4848cd6a257c23a95ee85f83c75302f97bcc4797f154b6a
 
-build_options_default="ffmpeg"
+build_options="ffmpeg gles2 glslang jack opengl pulseaudio qt5 sdl2 vulkan wayland x11"
+build_options_default="ffmpeg gles2 glslang opengl pulseaudio sdl2 vulkan wayland x11"
 
-case "$XBPS_TARGET_MACHINE" in
-	i686*|x86_64*|ppc64*|aarch64*) build_options_default+=" opengl pulseaudio sdl2 x11 vulkan";;
-	armv[67]*)
-		makedepends+=" rpi-userland-devel"
-		LDFLAGS="-L${XBPS_CROSS_BASE}/opt/vc/lib -lbcm_host"
-		;;
-esac
+if [ "$build_option_gles2" ]; then
+	configure_args+=" --enable-opengles --enable-opengles3"
+else
+	configure_args+=" --disable-opengles --disable-opengles3"
+fi
+
+if [ "$build_option_opengl" ]; then
+	if [ -z "$build_option_glslang" ]; then
+		msg_error "$pkgname: 'opengl' option requires 'glslang'.\n"
+	fi
+	configure_args+=" --enable-opengl_core"
+else
+	configure_args+=" --disable-opengl_core"
+fi
+
+if [ "$build_option_gles2" -o "$build_option_opengl" ]; then
+	configure_args+=" --enable-egl --enable-kms"
+	makedepends+=" MesaLib-devel"
+fi
+
+if [ "$build_option_vulkan" ]; then
+	if [ -z "$build_option_glslang" ]; then
+		msg_error "$pkgname: 'vulkan' option requires 'glslang'.\n"
+	fi
+fi
+
+if [ "$build_option_wayland" ]; then
+	if [ -z "$build_option_gles2" -o -z "$build_option_opengl" ]; then
+		msg_error "$pkgname: 'wayland' option requires 'gles2' or 'opengl'.\n"
+	fi
+fi
+
+if [ "$build_option_x11" ]; then
+	if [ -z "$build_option_gles2" -o -z "$build_option_opengl" ]; then
+		msg_error "$pkgname: 'x11' option requires 'gles2' or 'opengl'.\n"
+	fi
+fi
 
 do_configure() {
 	if [ "$CROSS_BUILD" ]; then
-		mkdir -p /opt/vc/lib
 		configure_args+=" --host=${XBPS_CROSS_TRIPLET}"
 	fi
 
 	case "$XBPS_TARGET_MACHINE" in
 		i686*|x86_64*) configure_args+=" --enable-sse --enable-threads";;
 		ppc64*) configure_args+=" --enable-threads";;
 		aarch64*) configure_args+=" --disable-neon";;
-		armv6*) configure_args+=" --disable-neon"
-			CFLAGS+=" -I${XBPS_CROSS_BASE}/opt/vc/include"
-			LDFLAGS+=" -L${XBPS_CROSS_BASE}/opt/vc/lib -Wl,-R /opt/vc/lib";;
-		armv7*) configure_args+=" --disable-neon --enable-threads"
-			CFLAGS+=" -I${XBPS_CROSS_BASE}/opt/vc/include"
-			LDFLAGS+=" -L${XBPS_CROSS_BASE}/opt/vc/lib -Wl,-R /opt/vc/lib";;
+		armv6*) configure_args+=" --disable-neon";;
+		armv7*) configure_args+=" --disable-neon --enable-threads";;
```

Comment:
It'd probably be reasonable to enable threads on aarch64 too, and leave NEON on, and enable threads on ppc32; do that by changing the `ppc64*` case to `ppc*|aarch64*`, then drop the standalone `aarch64*` case